### DISCUSSION TITLE
Trying to figure out all configuration options

### DISCUSSION THREAD

**Discussion by iKevinShah** [phpredis/phpredis#2066] [Q&A] [answered]
Is the a config / ini file where I can locate all the settings configurable and default values for phpredis?

I tried looking in source but at various points `ini_set` is being used (in tests/ )

couldnt find any sample `php.ini` file like php provides in its source. Googling and all was not helpful. 
I see various settings mentioned in different github issues but I am looking for all options and going through them. Any idea where to look for. 

It is possible I might have overlooked a link open in the README or similar file. Please point me in right direction.

Looking forward to hearing from you.

**Comment by michael-grunder** [comment, marked as answer]
You can probably glean most of what you'd want from this section in `redis.c`:  The first argument is the INI setting, and the second is the default value.

https://github.com/phpredis/phpredis/blob/0719c1eca0f3ce5650f51dfe878350147d424bbe/redis.c#L79-L115

That said, it may not be a bad idea to group this information somewhere in the documentation.

**Reply by iKevinShah** [reply]
That was quick and fantastic. I think this is what I was looking for, but was grepping `ini_set` instead of this. Thank you so much. Still one needs to make an educated guess as to what it does. But its a start.

**Comment by yatsukhnenko** [comment]
`php --ri redis`

**Reply by iKevinShah** [reply]
Thank you @yatsukhnenko , That helps. Same as earlier though, one needs to guess what it does, but yes, that at least is a step in right direction. Thank you once again.
